va/validation-authority.go

```diff
@@ -227,3 +231,40 @@ func (va ValidationAuthorityImpl) UpdateValidations(authz core.Authorization, ch
 	go va.validate(authz, challengeIndex)
 	return nil
 }
+
+// CheckCAA verifies that the indicated subscriber domain has a CAA record
+// authorizing the specified CA domain to issue a certificate.
+func (va *ValidationAuthorityImpl) CheckCAARecords(identifier core.AcmeIdentifier) (present, valid bool, err error) {
+	domain := strings.ToLower(identifier.Value)
+	caaSet, err := getCaaSet(domain, va.DNSResolver, va.DNSTimeout)
+
+	if caaSet == nil {
+		// No CAA records found, can issue
+		present = false
+		valid = true
+		return
+	} else if caaSet.criticalUnknown() {
+		present = true
+		valid = false
+		return
+	} else if len(caaSet.issue) > 0 || len(caaSet.issuewild) > 0 {
+		present = true
+		var checkSet []*CAA
+		if strings.SplitN(domain, ".", 2)[0] == "*" {
+			checkSet = append(caaSet.issuewild, caaSet.issue...)
```

Comment:
This seems to indicate that, to issue a wildcard, it's sufficient that the issuer domain be present in either issuewild or issue. Is that right? What if issuewild has foo.com and issue has bar.com?